### PR TITLE
chore!: replace dummy ABIs with `FunctionSignature` type alias

### PR DESCRIPTION
**Pull request by TomAFrench**
# Related issue(s)

# Description

## Summary of changes

Following #851, we're now exposing methods to compute an "ABI" which doesn't contain half the fields of an actual ABI (namely, the witness maps). We then need some reduced ABI-ish format which just tells us what the arguments/return values are and should look like.

This PR introduces this as a `FunctionSignature` type alias which just describes the arguments and return values of a function.

## Test additions / changes

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.
- [ ] This PR requires documentation updates when merged.

# Additional context

### CODE REVIEW

File: crates/nargo/src/cli/check_cmd.rs [hunk 1]
```diff
@@ -63,11 +64,26 @@ fn check_from_path<P: AsRef<Path>>(p: P, allow_warnings: bool) -> Result<(), Cli
...
+    let default_value = |typ: AbiType| {
+        if matches!(typ, AbiType::Array { .. }) {
+            "[]"
+        } else {
+            ""
+        }
+    };
+
+    let mut map =
+        btree_map(parameters, |AbiParameter { name, typ, .. }| (name, default_value(typ)));
```
Comment (jfecher) [inline crates/nargo/src/cli/check_cmd.rs:80]: If we're already factoring out the lambda, I think we should go all the way and include the whole lambda in `default_value`.

File: crates/nargo/src/cli/check_cmd.rs [hunk 2]
```diff
@@ -63,11 +64,26 @@ fn check_from_path<P: AsRef<Path>>(p: P, allow_warnings: bool) -> Result<(), Cli
     Ok(())
 }
 
-fn build_empty_map(abi: Abi) -> BTreeMap<String, &'static str> {
-    btree_map(abi.parameters, |AbiParameter { name, typ, .. }| {
-        let default_value = if matches!(typ, AbiType::Array { .. }) { "[]" } else { "" };
-        (name, default_value)
-    })
+fn build_empty_map(
```
Comment (jfecher) [inline crates/nargo/src/cli/check_cmd.rs:67]: Can we update this function name now? The map we're returning does not seem to be empty at all in the traditional sense. It also seems we're building something more specific than an arbitrary map.

File: crates/nargo/src/cli/check_cmd.rs (hunk 1 above)
Comment (TomAFrench) [reply crates/nargo/src/cli/check_cmd.rs:80]: This gets a bit mucky when it comes to dealing with the return value however. We'd have to pass in a dummy name just to immediately extract the default value term. This seems cleaner imo.

**Review by jfecher — APPROVED** [review]
👍

**Pull request merged**